include/silo/storage/database_partition.h

```diff
@@ -10,6 +10,10 @@
 #include "silo/storage/metadata_store.h"
 #include "silo/storage/sequence_store.h"
 
+namespace silo {
+struct Database;
+}
```

Comment:
Why do we need this here? Why do we need `friend class Database`? I'd actually expect that the partition does not depend on the db. Instead the db depends on the partition.

Reply:
Because of one private member access which I now made public. This is still not perfectly polished now (but it regards preprocessing), but I guess it is better than this backwards dependence that you noticed